src/fal/run_scripts.py

```diff
@@ -80,6 +90,37 @@ def ref_resolver(target_model_name: str, target_package_name: Optional[str] = No
     return ref_resolver
 
 
+def _get_firestore_writer(model: ParsedModelNode, manifest: Manifest):
+    # Use the application default credentials
+    cred = credentials.ApplicationDefault()
+    app_name = str(uuid.uuid4())
+
+    app = firebase_admin.initialize_app(cred, {
+        'projectId': model.database,
+    }, name=app_name)
+
+    db = firestore.client(app=app)
+
+    def firestore_writer(
+            df: pd.DataFrame,
+            collection: str,
+            document: str,
+            key_column: str = None,
+            merge=True):
+        doc_ref = db.collection(collection).document(document)
```

Comment:
Isn't document referring to a single document within the collection?
And then we send a lot of records to it? I am not understanding how this one works.

Reply:
The thinking here is that a collection refers to a set of dbt models and a document is a model. This way `doc_ref.get().to_dict()` will return the entire model as a Dict.

Reply:
It's now updated to map records to single documents